File: app/src/main/java/com/prance/teacher/features/check/view/CheckKeyPadAdapter.java
package com.prance.teacher.features.check.view;

import android.view.ViewGroup;

import com.prance.lib.teacher.base.ui.BaseLoadMoreAdapter;
import com.prance.lib.teacher.base.ui.BaseRecyclerAdapter;
import com.prance.lib.teacher.base.ui.BaseRecyclerHolder;
import com.prance.teacher.features.check.model.CheckKeyPadGroupTitle;

public class CheckKeyPadAdapter extends BaseLoadMoreAdapter<Object, BaseRecyclerHolder> {

    private static final int TITLE = 1;

    @Override
    public BaseRecyclerHolder onCreate(ViewGroup parent, int viewType) {
        if (viewType == 1) {
            return new CheckKeyPadTitleHolder(parent);
        } else {
            return new CheckKeyPadHolder(parent);
        }
    }

    @Override
    public void onBind(BaseRecyclerHolder viewHolder, int position, Object data) {
        viewHolder.onBind(data);
    }

    @Override
    protected int getNormalViewType(int position) {
        if (getData(position) instanceof CheckKeyPadGroupTitle) {
            return TITLE;
        }
        return super.getNormalViewType(position);
    }
}

File: app/src/main/java/com/prance/teacher/features/check/view/CheckKeyPadAdapter.kt
package com.prance.teacher.features.check.view

import android.text.Html

import com.chad.library.adapter.base.BaseMultiItemQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.chad.library.adapter.base.entity.MultiItemEntity
import com.prance.lib.database.KeyPadEntity
import com.prance.teacher.R
import com.prance.teacher.features.check.model.CheckKeyPadGroupTitle

/**
 * @author bingbing
 */
class CheckKeyPadAdapter(data: List<MultiItemEntity>) : BaseMultiItemQuickAdapter<MultiItemEntity, BaseViewHolder>(data) {

    companion object {
        const val TITLE_TYPE = 1
        const val ITEM_TYPE = 2
    }

    init {
        addItemType(TITLE_TYPE, R.layout.item_check_group_title)
        addItemType(ITEM_TYPE, R.layout.item_check_key_pad)
    }

    override fun convert(helper: BaseViewHolder, item: MultiItemEntity) {
        when (helper.itemViewType) {
            TITLE_TYPE -> {
                val bean = item as CheckKeyPadGroupTitle
                bean.run {
                    helper.setText(R.id.title, Html.fromHtml("""$title<font color="#3AF0EE">$number</font>个"""))
                }
            }
            ITEM_TYPE -> {
                val bean = item as KeyPadEntity
                bean.run {
                    helper.setText(R.id.keyNumber, keyId)
                    when (status) {
                        KeyPadEntity.OFFLINE -> helper.setImageResource(R.id.keyPadBackground, R.drawable.bg_keypad_background_offline)
                        else -> helper.setImageResource(R.id.keyPadBackground, R.drawable.bg_keypad_background_error)
                    }
                }
            }
        }
    }
}
